storage/src/adb/any/sync/client.rs

```diff
@@ -199,444 +250,598 @@ where
+                self.config.update_receiver = None;
+                Ok(StepResult::Continue(self))
+            }
+            SyncEvent::BatchReceived(fetch_result) => {
+                // Process the fetch result
+                self.handle_fetch_result(fetch_result)?;
 
-        info!(
-            old_target = ?config.target,
-            new_target = ?new_target,
-            "applying target update"
-        );
+                // Request operations in the sync range
+                self.request_operations().await?;
 
-        // Check if the existing log contains data in the updated sync range
-        let log_size = log
-            .size()
-            .await
-            .map_err(|e| Error::Adb(adb::Error::JournalError(e)))?;
+                // Apply operations that are now contiguous with the current log size
+                self.apply_operations().await?;
 
-        let log = if log_size <= new_target.lower_bound_ops {
-            // Log is stale (last element is before the new lower bound)
-            // Reinitialize the log with the new sync range
-            log.close()
-                .await
-                .map_err(|e| Error::Adb(adb::Error::JournalError(e)))?;
+                Ok(StepResult::Continue(self))
+            }
+        }
+    }
 
-            debug!(
-                log_size,
-                old_bounds = ?config.target,
-                new_bounds = ?new_target,
-                "log is stale, reinitializing"
+    /// Request batches of operations from the resolver.
+    async fn request_operations(&mut self) -> Result<(), Error> {
+        let target_size = self.config.target.upper_bound_ops + 1;
+
+        // Special case: If we don't have pinned nodes, we need to extract them from a proof
+        // for the lower sync bound.
+        if self.pinned_nodes.is_none() {
+            let start_loc = self.config.target.lower_bound_ops;
+            let resolver = self.config.resolver.clone();
+            self.outstanding_requests.add(
+                start_loc,
+                Box::pin(async move {
+                    let result = resolver
+                        .get_operations(target_size, start_loc, NZU64!(1))
+                        .await;
+                    IndexedFetchResult { start_loc, result }
+                }),
             );
-            Journal::<E, Fixed<K, V>>::init_sync(
-                config.context.clone().with_label("log"),
-                JConfig {
-                    partition: config.db_config.log_journal_partition.clone(),
-                    items_per_blob: config.db_config.log_items_per_blob,
-                    write_buffer: config.db_config.log_write_buffer,
-                    buffer_pool: config.db_config.buffer_pool.clone(),
-                },
-                new_target.lower_bound_ops,
-                new_target.upper_bound_ops,
-            )
+        }
+
+        // Calculate the number of requests to make
```

Comment:
nit: clarify that this is the _max_ number of requests we can make?  Your loop will make fewer if there are fewer gaps to fill.